npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -233,16 +234,18 @@ def requirements_for_path(requirements, path)
         # rubocop:disable Metrics/MethodLength
         def handle_yarn_lock_updater_error(error, yarn_lock)
           error_message = error.message
+
           # Invalid package: When package.json doesn't include a name or version
           # Local path error: When installing a git dependency which
           # is using local file paths for sub-dependencies (e.g. unbuilt yarn
           # workspace project)
-          sub_dep_local_path_err = "refers to a non-existing file"
-          if error_message.match?(INVALID_PACKAGE) ||
-             error_message.include?(sub_dep_local_path_err)
-            raise_resolvability_error(error_message, yarn_lock)
+          if error_message.match?(INVALID_PACKAGE_REGEX) ||
+             error_message.include?(SUB_DEP_LOCAL_PATH_TEXT)
+            error_handler.raise_resolvability_error(error_message, yarn_lock)
           end
 
+          error_handler.handle_error(error)
```

Comment:
We are raising error from handle_error if it is a known error we can catch. If not we are not handling the error since the existing error handling flow is going on.